device/include/ReadAnyGroup.h

```diff
@@ -509,6 +510,28 @@ namespace ChimeraTK {
       return {};
     }
 
+    // if update is available, peek into the queue to check whether a DiscardValueException will be read
+    auto id = notification_queue.front();
+    try {
+      // call to empty() necessary before call to front(), to gain ownership of front element
+      if(push_elements[id].getHighLevelImplElement()->_readQueue.empty()) {
+        // cannot place the call to empty() into the assert, since it must be executed also in release builds
+        assert(false);
+      }
+      push_elements[id].getHighLevelImplElement()->_readQueue.front();
+    }
+    catch(detail::DiscardValueException&) {
```

Comment:
Is the behavior correct in a release build? Aren't we going to waitAny without an update or at least some weird inconsistent state in that case?